src/cljs/rems/table2.cljs

```diff
@@ -0,0 +1,134 @@
+(ns rems.table2
+  (:require [clojure.string :as str]
+            [re-frame.core :as rf]
+            [rems.atoms :refer [close-symbol search-symbol sort-symbol]]
+            [rems.text :refer [text]]))
+
+(defn- flip [order]
+  (case order
+    :desc :asc
+    :desc))
+
+(defn- change-sort-order [old-column old-order new-column]
+  (if (= old-column new-column)
+    (flip old-order)
+    :asc))
+
+(rf/reg-event-db
+ ::toggle-sorting
+ (fn [db [_ spec sort-column]]
+   (update-in db [::sorting (:id spec)]
+              (fn [sorting]
+                (-> sorting
+                    (assoc :sort-column sort-column)
+                    (assoc :sort-order (change-sort-order (:sort-column sorting)
+                                                          (:sort-order sorting)
+                                                          sort-column)))))))
+
+(rf/reg-sub
+ ::sorting
+ (fn [db [_ spec]]
+   (or (get-in db [::sorting (:id spec)])
+       {:sort-order :asc
+        :sort-column (:default-sort-column spec)})))
+
+(rf/reg-event-db
+ ::set-filtering
+ (fn [db [_ spec filtering]]
+   (assoc-in db [::filtering (:id spec)] filtering)))
+
+(rf/reg-sub
+ ::filtering
+ (fn [db [_ spec]]
+   (get-in db [::filtering (:id spec)])))
+
+(rf/reg-sub
+ ::sorted-rows
+ (fn [[_ spec] _]
+   [(rf/subscribe [(:rows spec)])
+    (rf/subscribe [::sorting spec])])
+ (fn [[rows sorting] _]
+   (->> rows
+        (sort-by #(get-in % [(:sort-column sorting) :sort-value])
+                 (case (:sort-order sorting)
+                   :desc #(compare %2 %1)
+                   #(compare %1 %2))))))
+
+(defn- display-row? [row columns filters]
+  (some (fn [column]
+          (str/includes? (get-in row [(:key column) :filter-value])
+                         filters))
+        columns))
+
+(rf/reg-sub
+ ::sorted-and-filtered-rows
+ (fn [[_ spec] _]
+   [(rf/subscribe [::sorted-rows spec])
+    (rf/subscribe [::filtering spec])])
+ (fn [[rows filtering] [_ spec]]
+   (let [filters (str/lower-case (str (:filters filtering)))
+         columns (->> (:columns spec)
+                      (filter :filterable?))]
+     (->> rows
+          (map (fn [row]
+                 ;; performance optimization: hide DOM nodes instead of destroying them
+                 (assoc row ::display-row? (display-row? row columns filters))))))))
+
```

Comment:
I think I like how all the logic is in the subscriptions and only the views are left below.

But now this component isn't reusable, right? You can't have multiple table2s on the same page?

Reply:
I guess this is intended to be fixed in later PRs

Reply:
All the subcriptions are parameterized with a unique ID, so it should be reusable unless I have fundamentally misunderstood re-frame's subscriptions.